Fixed #25357 -- Database deduplication on Oracle.

aaugustin opened the pull request:
This branch is incorrectly named. At first I thought the problem would affect mirrors, in fact it only affects duplicate database configurations.

I extracted this bugfix when I noticed the problem while working on a branch for another feature (#4761). The first commit is needed in that branch.

aaugustin commented:
I cannot reproduce the test failures locally :-/ I'm attempting a blind fix...

aaugustin commented:
This should be good to go.

shaib commented:
The change in general seems to make sense (although I only gave it a brief look, can't say I followed all the details). There are two comments which come to mind:

- The name `describe_databases()` seems a bit generic. I would go for something like `find_mirroring_sources()`.
- The value returned from this method includes a list of databases with their mirroring aliases (as discovered by comparing signatures), and another dictionary of mirrors (derived using the `MIRROR` entries in `TEST` settings). Does the separation (still) make sense? Is it not better to just treat all mirrors alike?

aaugustin commented:
I renamed `describe_databases()` to `get_unique_databases_and_mirrors()` and improved its docstring a bit.

I also added another commit to remove the separation between "implicit mirrors" (same settings) and explicit mirrors.

aaugustin commented:
Test failures aren't related to the code, Jenkins couldn't checkout from GitHub.
